va/va_enc_hevc.h

```diff
@@ -45,6 +45,200 @@ extern "C" {
+         */
+        uint32_t separate_colour_planes     : 2;
+        /** Scaling lists.
+         *
+         * Allows scaling_list() elements to be present in both the SPS
+         * and the PPS.  The decoded form of the scaling lists must also
+         * be supplied in a VAQMatrixBufferHEVC buffer when scaling lists
+         * are enabled.
+         */
+        uint32_t scaling_lists              : 2;
+        /** Asymmetric motion partitions.
+         *
+         * Allows setting amp_enabled_flag in the SPS.
+         */
+        uint32_t amp                        : 2;
+        /** Sample adaptive offset filter.
+         *
+         * Allows setting slice_sao_luma_flag and slice_sao_chroma_flag
+         * in slice headers.
+         */
+        uint32_t sao                        : 2;
+        /** PCM sample blocks.
+         *
+         * Allows setting pcm_enabled_flag in the SPS.  When enabled
+         * PCM parameters must be supplied with the sequence parameters,
+         * including block sizes which may be further constrained as
+         * noted in the VAConfigAttribEncHEVCBlockSizes attribute.
+         */
+        uint32_t pcm                        : 2;
+        /** Temporal motion vector Prediction.
+         *
+         * Allows setting slice_temporal_mvp_enabled_flag in slice
+         * headers.
+         */
+        uint32_t temporal_mvp               : 2;
+        /** Strong intra smoothing.
+         *
+         * Allows setting strong_intra_smoothing_enabled_flag in the SPS.
+         */
+        uint32_t strong_intra_smoothing     : 2;
+        /** Dependent slices.
+         *
+         * Allows setting dependent_slice_segment_flag in slice headers.
+         */
+        uint32_t dependent_slices           : 2;
+        /** Sign data hiding.
+         *
+         * Allows setting sign_data_hiding_enable_flag in the PPS.
+         */
+        uint32_t sign_data_hiding           : 2;
+        /** Constrained intra prediction.
+         *
+         * Allows setting constrained_intra_pred_flag in the PPS.
+         */
+        uint32_t constrained_intra_pred     : 2;
+        /** Transform skipping.
+         *
+         * Allows setting transform_skip_enabled_flag in the PPS.
+         */
+        uint32_t transform_skip             : 2;
+        /** QP delta within coding units.
+         *
+         * Allows setting cu_qp_delta_enabled_flag in the PPS.  When
+         * enabled, diff_cu_qp_delta_depth must also be set to a value
+         * allowed by the VAConfigAttribEncHEVCBlockSizes attribute.
+         */
+        uint32_t cu_qp_delta                : 2;
+        /** Weighted prediction.
+         *
+         * Allows setting weighted_pred_flag and weighted_bipred_flag in
+         * the PPS.  The pred_weight_table() data must be supplied with
+         * every slice header when weighted prediction is enabled.
+         */
+        uint32_t weighted_prediction        : 2;
```

Comment:
Same with SAO. Driver also allows weighted_pred_flag and weighted_bipred_flag to be disabled/enabled respectively. How to set this field?

Reply:
So the same case but for hardware which only has partial weighted prediction support?  Again, I was imagining that no such hardware would exist, but we could separate them if you do have such hardware.  (Hardware which lacks support for encoding P or B frames wouldn't care about the value of the other flag, because it would never be used.)

Reply:
I have a question about weighted_prediction, media-driver support two kind of weighted-prediction: 1. explicit weighted prediction, application should provide pred_weight_table, and media-driver do weighted prediction basing on the pred_weight_table from applicaiton, so a pre-process needed (such as, fade in/out detection) before encoding  2. media driver based weighted prediction (or implicit weight prediction), media-driver will analyze the statistic datas and will provide a pred_weigt_table internally, to help the video quality. it happens inside media-driver during encoding process. 
my question is: how to report this kind of capability?

Reply:
@XinfengZhang  I think the setting for driver-produced prediction weights should be signalled by its own attribute rather than embedded here, since it's not actually related to the codec implementation - it would exist in H.264/H.266 as well.

(There are other similar cases where either a driver-internal process or some kind of preprocessing step could be used to produce values for codec processes - e.g. the segmentation map in VP8/VP9/AV1, scaling lists in H.264/H.265/H.266, global motion in AV1.  It would be sensible for these to all be treated in a consistent way.)